client/client.go

```diff
@@ -223,6 +229,31 @@ func WithChainInfo(chainInfo *chain.Info) Option {
 	}
 }
 
+// WithVerifiedResult provides a checkpoint of randomness verified at a given round.
+// Used in combination with `VerifyFullChain`, this allows for catching up only on
+// previously not-yet-verified results.
+func WithVerifiedResult(result Result) Option {
+	return func(cfg *clientConfig) error {
+		if cfg.previousResult != nil && cfg.previousResult.Round() > result.Round() {
+			return errors.New("refusing to override verified result with an earlier result")
+		}
+		return nil
+	}
+}
+
+// WithFullChainVerification validates random beacons not just as being generated correctly
+// from the group signature, but ensures that the full chain is deterministic by making sure
+// each round is derived correctly from the previous one. In cases of compromise where
+// a single party learns sufficient shares to derive the full key, malicious randomness
+// could otherwise be generated that is signed, but not properly derived from previous rounds
+// according to protocol.
+func WithFullChainVerification() Option {
```

Comment:
Shouldnt it be the other way around ? Like `DisableFullChainVerification` ? I don't even know if we should provide the option at all. Why should we not stick to the highest default standard ?

Reply:
the additional potential for full-chain fetch is potentially very expensive, especially since we don't have batch endpoints.
while in practice clients may do okay by putting in a recent point of trust, I'm not sure all users realize what they're signing up for in terms of computational complexity and bandwidth when they go to grab the latest randomness

Reply:
more generally, if we're switching to this as our default mode, we should re-visit our assumption about not needing historic batch-fetching, because i expect the client-server wire interface would want to change to support bulk range-fetches if this is a common thing clients are doing.

Reply:
> our assumption about not needing historic batch-fetching,

I don't recall it was  really an _explicit_ assumption of not having that so I'd be happy to add that for sure.

Anyway, I think since we don't have directly the use case case that uses this functionality, this will do - we may need to revisit this API after some time with some user feedback.